### PR TITLE
a number of proposed fixes to get this working on a fresh machine

### PR DESCRIPTION
**Pull request by matthewcornell**
I made the following proposed changes:
- flu_ar2/main.py: fixed `main()` to handle default `today_date` (None) - was getting "TypeError: fromisoformat: argument must be str"
- flu_ar2/README.md: added required `cd fl_ar2`
- requirements.txt: added missing requirements. NB: specified `numpy==1.26.4` to avoid incompatibility apparently introduced in numpy 2.*: ValueError: numpy.dtype size changed, may indicate binary incompatibility.
- .gitignore: added `/flu_ar2/output/`
- added .python-version to specify 3.12 (my 3.9 version was causing type hint problems)
- added Pipfile and Pipfile.lock to ease debugging on different machines

### CODE REVIEW

**Comment by matthewcornell** [comment]
At some point we should talk about if/how the eventual docker .sh script will glue py and r portions together (with the .csv file from sarix being passed explicitly to R). Etc.

**Comment by elray1** [comment]
Thanks, @matthewcornell .  This all seems fine to me, except that I think if it works for you it'd be better to resolve the pandas/numpy discrepancy by using the most recent version of pandas rather than downgrading numpy.

fwiw I think this issue comes from this pathway:
 - [reichlab/idmodels depends on reichlab/sarix](https://github.com/reichlab/idmodels/blob/b08602e37f1a4d5d8aed9caac5b28188583b23a5/requirements/requirements.txt#L165)
 - [reichlab/sarix depends on covidcast](https://github.com/elray1/sarix/blob/61a651c8ab5f09b0509a1cecbda4e7cefb42e8cc/setup.py#L18)
 - [covidcast depends on "pandas<2"](https://github.com/cmu-delphi/covidcast/blob/c89e4d295550ba1540d64d2cc991badf63ad04e5/Python-packages/covidcast-py/setup.py#L23)

However, covidcast is not actually used here.  I just used it to pull some data for a documentation/example in sarix, so a correct setup for sarix should not actually list it as a dependency for the package.  So, I think:
 - (a) it would be safe to override that pandas<2 requirement to install the latest pandas
 - (b) an actual solution could/should involve removing the covidcast dependency from sarix

**Comment by matthewcornell** [comment]
Thanks, @elray1 . How do you want to proceed?

**Comment by elray1** [comment]
One other comment on this PR (didn't notice this earlier): each model will have its own requirements, so I think the pipfile and `.python_version` should go within the model-specific folder

File: flu_ar2/README.md [hunk 1]
```diff
@@ -3,11 +3,13 @@
 To test this out locally, run the following with `flu_ar2` as your working directory.  Note that somewhere along the line, incompatible versions of pandas and numpy ended up in a requirements file somewhere.  I have not dug into where or why that happened.  Here we manually override the pandas installation.
 
 ```
+cd <this_repo>
```
Comment (elray1) [inline flu_ar2/README.md:6]: noting that instructions above say to run all of the below with `flu_ar2` as the working directory.  the idea i had with this was that we may want to allow for a separate set of requirements per model.  in practice, though, for now the requirements will be largely the same across models...

File: flu_ar2/main.py [hunk 2]
```diff
@@ -16,7 +16,10 @@
 )
 def main(today_date: str | None = None):
     """Get clade counts and save to S3 bucket."""
-    today_date = datetime.date.fromisoformat(today_date)
+    try:
+        today_date = datetime.date.fromisoformat(today_date)
+    except (TypeError, ValueError):  # if today_date is None or a bad format
+        today_date = datetime.date.today()
```
Comment (elray1) [inline flu_ar2/main.py:22]: another option, which i am taking in the next model, is just to require that the `today_date` argument be passed?

**Pull request closed without merging**